pkg/controller/controller.go

```diff
@@ -476,16 +561,31 @@ func (c *controller) updateBrokerFinalizers(
 // Service class handlers and control-loop
 
 func (c *controller) serviceClassAdd(obj interface{}) {
-	serviceClass, ok := obj.(*v1alpha1.ServiceClass)
-	if serviceClass == nil || !ok {
+	key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
+	if err != nil {
+		glog.Errorf("Couldn't get key for object %+v: %v", obj, err)
 		return
 	}
+	c.serviceClassQueue.Add(key)
+}
 
-	c.reconcileServiceClass(serviceClass)
+func (c *controller) reconcileServiceClassKey(key string) error {
+	serviceClass, err := c.serviceClassLister.Get(key)
+	if errors.IsNotFound(err) {
+		glog.Infof("ServiceClass has been deleted %v", key)
```

Comment:
Make this log message more explanatory.  It's tough to know what this does as someone reading the log who hasn't read the controller code.